src/macro-expansion.md

```diff
@@ -1 +1,66 @@
 # Macro expansion
+
+Macro expansion happens during parsing. `rustc` has two parsers, in fact: the
+normal Rust parser, and the macro parser. During the parsing phase, the normal
+Rust parser will call into the macro parser when it encounters a macro. The
```

Comment:
can you be more precise about what a reference to a macro is? e.g. ,do you mean a macro invocation, like `foo!(...)`?

Reply:
Also, is it really called from the *parser*? I thought there was a second phase that came after parsing, but maybe I'm going to learn something here =)

Reply:
Let me verify that :)

Reply:
@nikomatsakis 

Ok, so it looks like
- the macro parser is called from `macro_rules::compile` https://github.com/rust-lang/rust/blob/ca9cf3594ab25d2809ac576dfc9defb8e87b45b8/src/libsyntax/ext/tt/macro_rules.rs#L184 which transforms a macro invocation into a syntax extension.
- `macro_rules::compile` is called from `librustc_resolve/macros.rs` and `librustc_resolve/build_reduced_graph.rs`... which I'm guessing is doing name resolution? Does this run after the parser?